httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CacheUpdateHandler.java

```diff
@@ -123,18 +128,29 @@ public HttpCacheEntry updateParentCacheEntry(
         return new HttpCacheEntry(
                 src.getRequestInstant(),
                 src.getResponseInstant(),
+                src.getRequestMethod(),
+                src.getRequestURI(),
+                headers(src.requestHeaderIterator()),
                 src.getStatus(),
-                src.getHeaders(),
+                headers(src.headerIterator()),
                 resource,
                 variantMap);
     }
 
-    private Header[] mergeHeaders(final HttpCacheEntry entry, final HttpResponse response) {
-        if (DateSupport.isAfter(entry, response, HttpHeaders.DATE)) {
-            return entry.getHeaders();
+    private static HeaderGroup headers(final Iterator<Header> it) {
+        final HeaderGroup headerGroup = new HeaderGroup();
+        while (it.hasNext()) {
+            headerGroup.addHeader(it.next());
         }
+        return headerGroup;
+    }
+
+    private HeaderGroup mergeHeaders(final HttpCacheEntry entry, final HttpResponse response) {
```

Comment:
Currently, the method iterates over the response headers twice: once to remove matching headers from the cache, and once to add the response headers. In both iterations, the Content-Encoding and Content-Length headers are being skipped. This seems inefficient.

Reply:
@arturobernalg Bear with me. `Content-Length ` stuff will be gone soon.